src/inabox/inabox-resources.js

```diff
@@ -56,10 +73,21 @@ export class InaboxResources {
     /** @const @private {!Deferred} */
     this.firstPassDone_ = new Deferred();
 
+    /** @private {?IntersectionObserver} */
+    this.intersectionObserver_ = null;
```

Comment:
Be explicit. Sharing InOb instances for any generic reason is hard. Let's call it `inViewportObserver`.